sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -286,27 +286,36 @@ object Klaviyo {
      *
      * @param intent the [Intent] from opening a notification
      */
-    fun handlePush(intent: Intent?) = safeApply(preInitQueue) {
-        if (intent?.isKlaviyoNotificationIntent != true) {
-            Registry.log.verbose("Non-Klaviyo intent ignored")
-            return@safeApply
-        }
-
-        val event = Event(EventMetric.OPENED_PUSH)
-        val extras = intent.extras
+    fun handlePush(intent: Intent?): Klaviyo = this
```

Comment:
This is the primary purpose of the PR: if the developer has registered a deep link handler, we should invoke it when they call `handlePush`. 
I made it independent of the  event enqueuing (so that a failure there doesn't affecting linking) but still filtered to pushes originating from klaviyo only.